custom_components/kia_uvo/KiaUvoApiEU.py

```diff
@@ -424,7 +424,13 @@ def get_cached_vehicle_status(self, token: Token):
         response = requests.get(url, headers=headers)
         response = response.json()
         _LOGGER.debug(f"{DOMAIN} - get_cached_vehicle_status response {response}")
-        return response["resMsg"]["vehicleStatusInfo"]
+        if response["resCode"] == 0000:
+            return response["resMsg"]["vehicleStatusInfo"]
+        else:
+            _LOGGER.debug(
+                f"{DOMAIN} - get_cached_vehicle_status API failed to respond with data"
+            )
+            return None
```

Comment:
returning None would generate similar error in upstream functions, will it? I might be missing something.